2019-03-Autumn/5_graphs/find_mutual_friends/Vertex.java

```diff
@@ -0,0 +1,35 @@
+import java.util.Set;
+
+public class Vertex {
+    private String name;
+    private Set<Vertex> friends;
+
+    public Vertex(String name, Set<Vertex> friends) {
+        this.name = name;
+        this.friends = friends;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Set<Vertex> getFriends() {
+        return friends;
+    }
+
+    public void setFriends(Set<Vertex> friends) {
+        this.friends = friends;
+    }
+
+    @Override
+    public String toString() {
+        return "Vertex{" +
+                "name='" + name + '\'' +
+                ", friends=" + friends +
+                '}';
+    }
```

Comment:
I added a commit which fixes the StackOverflowError for this method when it gets called, and added calls to the tests to print the vertex so users can visually see what each vertex contains.

Reply:
Thank you! You're right, that would be helpful if print statements didn't blow up...